### PR TITLE
Fix ChangesZone triggers with Defined$

### PR DESCRIPTION
**Pull request by tool4ever**

### CODE REVIEW

File: forge-gui/res/cardsfolder/a/angel_of_condemnation.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ PT:3/3
 K:Flying
 K:Vigilance
 A:AB$ ChangeZone | Cost$ 2 W T | ValidTgts$ Creature.Other | Mandatory$ True | TgtPrompt$ Select another target creature | Origin$ Battlefield | Destination$ Exile | RememberChanged$ True | SubAbility$ DelTrig | SpellDescription$ Exile another target creature. Return that card to the battlefield under its owner's control at the beginning of the next end step.
-SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigBounce | RememberObjects$ RememberedLKI | TriggerDescription$ Return exiled creature to the battlefield. | SubAbility$ DBCleanup
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ TrigBounce | RememberObjects$ RememberedLKI | TriggerDescription$ Return the card to the battlefield. | SubAbility$ DBCleanup
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/a/angel_of_condemnation.txt:8]: might be nice with `TriggerDescription$ Return {c:DelayTriggerRememberedLKI} to the battlefield under its owner's control at the beginning of the next end step.` but I don't know if that token would work yet

File: forge-ai/src/main/java/forge/ai/ability/CountersPutAi.java [hunk 2]
```diff
@@ -362,7 +362,7 @@ public boolean apply(Card input) {
             }
         }
 
-        if ("Fight".equals(logic)) {
+        if ("Fight".equals(logic) || "PowerDmg".equals(logic)) {
```
Comment (Northmoc) [inline forge-ai/src/main/java/forge/ai/ability/CountersPutAi.java:365]: For Nissa's Judgment?

File: forge-gui/res/cardsfolder/f/frenetic_sliver.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ PT:2/2
 S:Mode$ Continuous | Affected$ Sliver | AddAbility$ Frenetic | AddSVar$ DBExile & DelTrig & MoveBack & DBSacSelf | Description$ All Slivers have "{0}: If this permanent is on the battlefield, flip a coin. If you win the flip, exile this permanent and return it to the battlefield under its owner's control at the beginning of the next end step. If you lose the flip, sacrifice it."
 SVar:Frenetic:AB$ FlipACoin | Cost$ 0 | ConditionPresent$ Card.Self | ConditionCompare$ EQ1 | WinSubAbility$ DBExile | LoseSubAbility$ DBSacSelf | SpellDescription$ If this permanent is on the battlefield, flip a coin. If you win the flip, exile this permanent and return it to the battlefield under its owner's control at the beginning of the next end step. If you lose the flip, sacrifice it.
 SVar:DBExile:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | Defined$ Self | SubAbility$ DelTrig
-SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ MoveBack | Static$ True
+SVar:DelTrig:DB$ DelayedTrigger | Mode$ Phase | Phase$ End of Turn | Execute$ MoveBack
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/f/frenetic_sliver.txt:8]: `TriggerDescription$ Return CARDNAME to the battlefield under its owner's control at the beginning of the next end step.` ?

File: forge-ai/src/main/java/forge/ai/ability/CountersPutAi.java (hunk 2 above)
Comment (Northmoc) [reply forge-ai/src/main/java/forge/ai/ability/CountersPutAi.java:365]: Ah, now I see Hunter's Edge as well – nice

**Review by Northmoc — COMMENTED** [review]
Looks great! I think it's going to make you request another review after any changes with those TriggerDescriptions even if I approve it now.

File: forge-gui/res/cardsfolder/g/generals_regalia.txt [hunk 4]
```diff
@@ -5,7 +5,7 @@ A:AB$ ChooseSource | Cost$ 3 | Choices$ Card,Emblem | AILogic$ NeedsPrevention |
 SVar:DBEffect:DB$ Effect | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control to redirect damage to | ReplacementEffects$ GeneralDamage | Triggers$ OutOfSight | RememberObjects$ Targeted | SubAbility$ DBCleanup
 SVar:GeneralDamage:Event$ DamageDone | ValidSource$ Card.ChosenCard,Emblem.ChosenCard | ValidTarget$ You | DamageTarget$ Remembered | ReplaceWith$ GeneralDmg | Description$ The next time a source of your choice would deal damage to you this turn, that damage is dealt to target creature you control instead.
 SVar:GeneralDmg:DB$ ReplaceEffect | VarName$ Affected | VarValue$ Remembered | VarType$ Card | SubAbility$ ExileEffect
-SVar:OutOfSight:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | Defined$ Imprinted | Execute$ ExileEffect | Static$ True
+SVar:OutOfSight:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Card.IsImprinted | Execute$ ExileEffect | Static$ True
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/g/generals_regalia.txt:8]: i might be wrong, but this card doesn't seems to use imprinted anyway?

**Review by Northmoc — APPROVED** [review]
👍

**Pull request merged**